windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas

```diff
@@ -753,26 +789,159 @@ procedure TMergeKeyboardInfo.CheckOrAddMinKeymanVersion;
+    end;
+  finally
+  end;
 
+  // Parse targets to populate platformSuppport
   v := TJSONObject.Create;
 
-  if Assigned(FKMPInfFile) then
+  for keyboardFile in FPackageKMXFileInfos do
   begin
-    v.AddPair('windows', 'full');
-    v.AddPair('macos', 'full');
-  end
-  else if Length(FKMXFiles) > 0 then
-    v.AddPair('windows', 'full');
+    targets := StringToKeymanTargets(keyboardFile.Info.Targets);
+    if ktAny in targets then
+    begin
+      AddNewPair('windows', 'full');
+      AddNewPair('macos', 'full');
+      AddNewPair('linux', 'full');
+      AddNewPair('desktopWeb', 'full');
+      AddNewPair('mobileWeb', 'full');
+      AddNewPair('android', 'full');
+      AddNewPair('ios', 'full');
+    end
+    else
+    begin
+      // If targets empty, then include all desktop targets
+      if (targets = []) then
+      begin
+        AddNewPair('windows', 'full');
+        AddNewPair('macos', 'full');
+        AddNewPair('linux', 'full');
+      end;
+
+      for target in targets do
+      begin
+        if target = ktDesktop then
+        begin
+          AddNewPair('windows', 'full');
+          AddNewPair('macos', 'full');
+          AddNewPair('linux', 'full');
+        end;
+        if target = ktWindows then
+          AddNewPair('windows', 'full');
+        if target = ktMacosx then
+          AddNewPair('macos', 'full');
+        if target = ktLinux then
+          AddNewPair('linux', 'full');
+
+        // FPackageKMXFileInfos can contain target information for web/mobile targets.
+        // This is a current limitation of FPackageJSFileInfos if there's no kmx files
+        if target = ktWeb then
+        begin
+          AddNewPair('desktopWeb', 'full');
+          AddNewPair('mobileWeb', 'full');
+        end;
+        if (target = ktMobile) then
+        begin
+          AddNewPair('android', 'full');
+          AddNewPair('ios', 'full');
+        end;
+        if (target = ktIphone) or (target = ktIpad) then
+        begin
+          AddNewPair('ios', 'full');
+        end;
+        if (target = ktAndroidphone) or (target = ktAndroidtablet) then
+        begin
+          AddNewPair('android', 'full');
+        end;
+      end;
+    end;
+  end;
 
+  for keyboardFile in FPackageJSFileInfos do
+  begin
+    targets := StringToKeymanTargets(keyboardFile.Info.Targets);
+    for target in targets do
+    begin
+      if (target = ktWeb) then
+      begin
+        AddNewPair('desktopWeb', 'full');
+        AddNewPair('mobileWeb', 'full');
+      end;
+      if (target = ktMobile) then
+      begin
+        AddNewPair('android', 'full');
+        AddNewPair('ios', 'full');
+      end;
+      if (target = ktIphone) or (target = ktIpad) then
+      begin
+        AddNewPair('ios', 'full');
+      end;
+      if (target = ktAndroidphone) or (target = ktAndroidtablet) then
+      begin
+        AddNewPair('android', 'full');
+      end;
+    end;
+  end;
+
+  // Handle JS file not in kmp
   if FJsFile <> '' then
   begin
```

Comment:
We probably need to keep android and ios in there until we complete the addition of all keyboard .js to packages in the repository (including legacy keyboards).